src/test/java/io/confluent/connect/hdfs/TestWithSecureMiniDFSCluster.java

```diff
@@ -53,18 +54,36 @@
 public class TestWithSecureMiniDFSCluster extends HdfsSinkConnectorTestBase {
 
   private static File baseDir;
+  private static FileSystem fs;
+  private static MiniDFSCluster cluster;
   private static String hdfsPrincipal;
   private static MiniKdc kdc;
   private static String keytab;
   private static String spnegoPrincipal;
   private static String connectorPrincipal;
   private static String connectorKeytab;
 
-  private MiniDFSCluster cluster;
-  private FileSystem fs;
-
   @BeforeClass
-  public static void initKdc() throws Exception {
+  public static void setup() throws Exception {
+    initKdc();
+
+    cluster = createDFSCluster();
+    fs = cluster.getFileSystem();
+  }
+
+  @AfterClass
+  public static void cleanup() throws IOException {
+    if (fs != null) {
+      fs.close();
+    }
+    if (cluster != null) {
+      cluster.shutdown(true);
+    }
+    UserGroupInformation.reset();
```

Comment:
We should reset the UGI after the non-kerberos tests as well, that way we don't have to rely on the order `kerb-tests -> regular tests`.